### ISSUE TITLE
Using asyncio with searches

### ISSUE THREAD

**Issue by mavam** [MISP/PyMISP#324]
I was wondering if someone could provide an example on how to use `asyncio` with PyMISP. I already got the 0mq subscription working, but not the plain API calls.

Let's say I have connected to an instance `misp = PyMISP(...)` and would like to invoke `misp.search(..)`. My main looks like this:

```python

def callback(*args):
    ...

async def main():
    misp = PyMISP(...)
    misp.search(async_callback=callback)

if __name__ == '__main__':
    asyncio.run(main())
```

If `search` was declared as `async`, I could loop until the search is done, e.g.:

```python
while searching:
    result = await misp.search(...)
    process(result)     
    if result.done:
         searching = False
```

Any ideas would be much appreciated.

**Comment by Rafiot** [comment]
Yeah, it isn't properly supported yet, and I want to rewrite it completely using pure python 3.6+.

The current code using async is probably somewhat broken and I cannot promise you it will not change in the coming months.

**Comment by mavam** [comment]
That's good to hear. Fortunately API stability is not a concern for us at the moment. I'm excited to see better asynchrony support.

But, what do you mean by "The current code using async is probably somewhat broken"? Is it not recommended to use `async_callback` at all?

**Comment by Rafiot** [comment]
I didn't use it in over 6 months and it is not tested *at all* so I cannot promise it works reliably. Using it won't break anything that I know of, but it may just randomly not work (and as you noticed the async callbacks aren't supported everywhere).

Note that It will only be available if you use python 3.6+ and I'm planning to add it on top of ExpandedPyMISP (in pymisp/aping.py). Until now, we didn't have much requests for async support, so if you want to contribute code that make your life easier, we are always welcoming PRs :)

**Comment by mavam** [comment]
Okay, so'll keep using the synchronous API for now.

I wish I had the cycles to add the async support myself, but I unfortunately have to wait...
